### DISCUSSION TITLE
Installation Script: ModuleNotFoundError: No module named 'pdm'

### DISCUSSION THREAD

**Discussion by erik-schulze** [searxng/searxng#4137] [Q&A] [answered]
**Version of SearXNG, commit number if you are using on master branch and stipulate if you forked SearXNG**
master branch, commit ID: 73e395c8ceecf015e1fa3c56b8c8e09780472604

**How did you install SearXNG?**
- the installation script, using the official guide
- Ubuntu 24.04

**What happened?**
- Python is throwing errors like "ModuleNotFoundError: No module named 'pdm'" after executing "pip install --use-pep517 --no-build-isolation -e ."

The redis configuartion section looks like:
```
Redis DB
========
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'searx'

In your instance, redis DB connector is configured at:

Traceback (most recent call last):
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'searx'
ERROR: SearXNG instance can't connect redis DB / check redis & your settings

```

**How To Reproduce**
- just install by script

**Comment by kokomo123** [comment]
I can also replicate this error with Debian 12. This broke me trying to upgrade from December 1st, to December 25th commit. I also tried using the docker container but that has its own cans of worms, even while dropping the cap line for the service. 
I also tried installing python3-pdm, but that didn't work either considering its in own pyenv.

**Comment by return42** [comment, marked as answer]
> I also tried installing python3-pdm, but that didn't work either considering its in own pyenv.

Don't install any additional packages into the virtual env of the SearXNG installation.

--> [pdm is not in the requirements-chain of SearXNG.](https://github.com/searxng/searxng/issues/4015#issuecomment-2564282420)
